vega_sim/wallet/vega_wallet.py

```diff
@@ -115,23 +116,32 @@ def generate_keypair(
             json=req,
         )
         response.raise_for_status()
-        return response.json()["key"]["pub"]
+        return list(response.json()["key"]["pub"])
 
-    def get_keypairs(self, wallet_name: str) -> Dict:
+    def get_keypairs(self, wallet_name: str) -> list:
         headers = {"Authorization": f"Bearer {self.login_tokens[wallet_name]}"}
         response = requests.get(
             WALLET_KEY_URL.format(wallet_server_url=self.wallet_url), headers=headers
         )
         response.raise_for_status()
-        return response.json()["keys"]
+        return {key["name"]: key["pub"] for key in response.json()["keys"]}
```

Comment:
This is still a dict, so doesn't need the type change above

Reply:
> Mostly looks reasonable, a couple of small changes.
> 
> One thing I think we probably should change is that returning the first key from the dictionary turned into a list feels a little fragile (it could change easily). Could we instead have some 'default' name which could be a fixed global variable then set/read that?

Ah that's a nice idea, I'll use "Key 1" as the default as I think that's the default the `vega wallet create` command creates. That will give us some crossover with existing wallets as well (in most cases if you are using an existing wallet you will have to pass a key name though).

Reply:
Not sure I understand sorry? This line reformats the dict using a dict comprehension no?

Reply:
It reformats the dict indeed but (unless I'm missing something) the function is still returning a different type of dict, not a list?

Reply:
Ah I am with you now sorry, forgot I changed the typing at the top! I will change that.